lib/core/src/Cardano/Wallet/Primitive/Types.hs

```diff
@@ -412,6 +427,17 @@ instance Buildable Direction where
         Outgoing -> "outgoing"
         Incoming -> "incoming"
 
+instance FromText Direction where
+    fromText txt = case txt of
+        "Outgoing" -> Right Outgoing
+        "Incoming" -> Right Incoming
+        _ ->
+            Left . TextDecodingError $ "not a valid value: " <> show txt
```

Comment:
A better error message would be to point out to what are the expected values (as we do in other places).

Reply:
done